aspnetcore/mvc/views/overview.md

```diff
@@ -1,228 +1,336 @@
+Set the title using `ViewBag` and the description using `ViewData` at the top of an *About.cshtml* view:
+
+```cshtml
+@{
+    Layout = "/Views/Shared/_Layout.cshtml";
+    ViewBag.Title = "About Contoso";
+    ViewData["Description"] = "Let us tell you about Contoso's philosophy and mission.";
+}
+```
 
-<!-- literal_block {"ids": [], "linenos": false, "xml:space": "preserve", "language": "html", "highlight_args": {"hl_lines": [13, 16, 17, 18]}} -->
+Read the properties but reverse the use of `ViewData` and `ViewBag`. In the *_Layout.cshtml* file, obtain the title using `ViewData` and obtain the description using `ViewBag`:
+
+```cshtml
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>@ViewData["Title"]</title>
+    <meta name="description" content="@ViewBag.Description">
+    ...
+```
+
+Remember that strings don't require a cast for `ViewData`. You can use `@ViewData["Title"]` without casting.
+
+Using both `ViewData` and `ViewBag` at the same time works, as does mixing and matching reading and writing the properties. The following markup is rendered:
 
 ```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>About Contoso</title>
+    <meta name="description" content="Let us tell you about Contoso's philosophy and mission.">
+    ...
+```
+
+**Summary of the differences between ViewData and ViewBag**
+
+* `ViewData`
+  * Derives from [ViewDataDictionary](/aspnet/core/api/microsoft.aspnetcore.mvc.viewfeatures.viewdatadictionary), so it has dictionary properties that can be useful, such as `ContainsKey`, `Add`, `Remove`, and `Clear`.
+  * Keys in the dictionary are strings, so whitespace is allowed. Example: `ViewData["Some Key With Whitespace"]`
+  * Any type other than a `string` must be cast in the view to use `ViewData`.
+* `ViewBag`
+  * Derives from [DynamicViewData](/aspnet/core/api/microsoft.aspnetcore.mvc.viewfeatures.internal.dynamicviewdata), so it allows the creation of dynamic properties using dot notation (`@ViewBag.SomeKey = <value or object>`), and no casting is required. The syntax of `ViewBag` makes it quicker to add to controllers and views.
+  * Simpler to check for null values. Example: `@ViewBag.Person?.Name`
+
+**When to use ViewData or ViewBag**
+
+Both `ViewData` and `ViewBag` are equally valid approaches for passing small amounts of data among controllers and views. The choice of which one to use (or both) comes down to personal preference or the preference of your organization. Generally, developers are consistent in their use of one or the other. They either use `ViewData` everywhere or use `ViewBag` everywhere, but you're welcome to mix and match them. Since both are dynamically resolved at runtime and thus prone to causing runtime errors, use them carefully. Some developers avoid them completely.
```

Comment:
"Generally, developers are consistent in their use of one or the other. They either use ViewData everywhere or use ViewBag everywhere, but you're welcome to mix and match them." > "Though you can mix and match ViewData and ViewBag objects, the code is easier to read and maintain when you choose only one to use consistently. "